### ISSUE TITLE
Assertion failed on parsing malformed SQL file (main/entry.c:1108)

### ISSUE THREAD

**Issue by kcwu** [universal-ctags/ctags#538]
How to reproduce
```
$ cat min.txt
0000000: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
0000010: 3030 3030 3030 3030 2063 7572 736f 7220  00000000 cursor
0000020: 2700 8000 3030 3030 3030 3030 3030 3030  '...000000000000
0000030: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
0000040: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
0000050: 3030 3030 23e2 0927 7f02 006e 7320 4020  0000#..'...ns @
0000060: 6172 7261 2013 6868 6868 6868 6868 6868  arra .hhhhhhhhhh
0000070: 6868 6868 6868 6868 6868 6868 6801 7f0a  hhhhhhhhhhhhh...
0000080: 2061 2b2b 2b2b 6c64 800a 0028 006e 7320   a++++ld...(.ns
0000090: 2720 6172 7261 4848 4848 4848 1301 7f0a  ' arraHHHHHH....
00000a0: 0000 0a20 6172 7261 2013 017f 0a00 6e73  ... arra .....ns
00000b0: 2027 205f 6f6e 6365 1548 7248 0a01 140a   ' _once.HrH....
00000c0: 0114 007f 0a0a 0909 277f 1700 5424 0a09  ........'...T$..
00000d0: 4191 0200 7873 2007 2061 727f 7320 0a48  A...xs . ar.s .H
00000e0: 4848 4848 0050 796d 65cf 0a01 0000 0010  HHHH.Pyme.......
00000f0: 730a 6c65 0a0a 007f 1b24 7401 007f ff00  s.le.....$t.....
0000100: 008c 70e0 8665 476f 6e63 6515 4872 200a  ..p..eGonce.Hr .
0000110: 2727 0000 0020 6153 6277 4c72 7261 2013  ''... aSbwLrra .
0000120: 017f 0a20 6172 722b 0f2b 1616 b716 1616  ... arr+.+......
0000130: 1616 1616 1616 1616 1616 1616 1616 1640  ...............@
0000140: 002b 2b2b 2b64 7401 007f ff00 0000 0000  .++++dt.........
0000150: 2061 5362 774c 7272 6120 1301 5f0a 2061   aSbwLrra .._. a
0000160: 7272 2b20 2720                           rr+ '
$ xxd -r < min.txt > min.sql
$ ctags min.sql
ctags: main/entry.c:1108: void initTagEntryFull(tagEntryInfo *const, const char *const, unsigned long, const char *, fpos_t, const char *): Assertion `name == NULL || strchr (name, '\t') == NULL' failed.
ctags: main/entry.c:1108: parsing min.sql:1 as SQL
Aborted
```

this is found by afl-fuzz

**Comment by masatake** [comment]
Simplified
```
00000000: 6375 7273 6f72 2027 090a                 cursor '..
```

**Comment by masatake** [comment]
Solved by #582.
